work/worker.go

```diff
@@ -722,20 +733,39 @@ func (env *Task) ApplyTransactions(txs *types.TransactionsByPriceAndNonce, bc Bl
 CommitTransactionLoop:
 	for atomic.LoadInt32(&abort) == 0 {
 		// Retrieve the next transaction and abort if all done
-		tx := txs.Peek()
-		if tx == nil {
+		if len(incorporatedTxs) == 0 {
 			// To indicate that it does not have enough transactions for params.BlockGenerationTimeLimit.
 			if numTxsChecked > 0 {
 				usedAllTxsCounter.Inc(1)
 			}
 			break
 		}
+
+		var (
+			tx      *types.Transaction
+			err     error
+			logs    []*types.Log
+			txOrGen = incorporatedTxs[0]
+			from    common.Address
+		)
+		switch txOrGen.(type) {
+		case *types.Transaction:
+			tx = txOrGen.(*types.Transaction)
+		case builder.TxGenerator:
+			txGen := txOrGen.(builder.TxGenerator)
+			nodeNonce := env.state.GetNonce(rewardbase)
+			tx, err = txGen(nodeNonce)
+			if tx == nil {
+				logger.Warn("TxGenerator returned a nil tx", "error", err)
+				continue
```

Comment:
We should pop `txGen`, otherwise it can fall into an infinite loop

Reply:
I see. However, since the target bundle cannot be identified, is it difficult to pop it here?
We need a method to compare the interface with the bundle.

Reply:
This allows us to put this processing before tx creation.
https://github.com/kaiachain/kaia/pull/272/files#diff-aa53dfe64c81ad7f4673af6857ca8ae7873bb18224202adb09f765ce4f458411R782-R788

Reply:
It was fixed at c866a31. Thank you.

Reply:
I don't think DeepEqual is ok. https://go.dev/play/p/B9E5K9I6p-u

Reply:
Because we refresh `bundles` every loop, I think it's safe to assume:
- TxGenerator must be in `bundles[0]`
- If tx is in bundle, it must be in `bundles[0]`. So it's sufficient to check `bundles[0].Has(tx)`

Reply:
On, nvm. For a sec I thought `bundles` was sorted.
Now that we update bundles' `TargetTxHash`, maybe we can find and use the bundle whose `TargetTxHash` is zero.

Reply:
That's good idea! I'll try it.

Reply:
I added a hash to TxGenerator to make it comparable. PTAL 6618e63.
There is room for refactoring, such as making TxOrGen(interface{}) a type, but this will be done in a separate PR.